pkg/distributor/distributor.go

```diff
@@ -133,6 +136,9 @@ type Distributor struct {
 	metadataValidationMetrics *validation.MetadataValidationMetrics
 
 	PushWithMiddlewares push.Func
+
+	// Pool of []byte used when marshalling write requests.
+	writeRequestBytePool sync.Pool
```

Comment:
I don't see why it would need to be a global variable, when we have a logical place to put it. I understand that's not always the case, but I think it fits into distributor nicely.